### DISCUSSION TITLE
Angular components in different directory

### DISCUSSION THREAD

**Discussion by chris-clarkson** [storybookjs/storybook#24878] [Help]
Hi, 

I have found that when writing stories, the component must be in the same directory as the story itself. If I reference a component from a different folder, none of the arguments/documentation is generated.

Is it possible to have components located in a different folder from the story?

Thanks,

**Comment by valentinpalkovic** [comment]
Can you provide a reproduction?

**Comment by chris-clarkson** [comment]
I have looked at this a bit further and it works if the component is anywhere under the src/ directory. If you move it to a folder that is not a child directory of src then it no longer works correctly. 

Hopefully that is enough information to replicate the issue your side

**Reply by Marklb** [reply]
It sounds like the problem is with how compodoc works. Without knowing where the outside file is and what your `tsconfig` is, it is hard to say for sure. If you could provide those or maybe a project with the problem then it may be easier to say what exactly is going on. Storybook doesn't generate the documentation, it just passed your `tsconfig` to [compodoc](https://github.com/compodoc/compodoc), which outputs a `documentation.json` that Storybook imports.

I have not gone through the compodoc code in a while and I don't see an explanation of how exactly the file searching works, other than that it uses the `tsconfig`. Here are a few things you could check:

- Is your file compiled as part of your project or another project?
  - If it is a component compiled separately then compodoc is probably not going to be able to parse the documentation info.
- Are you using a monorepo or workspace, with multiple projects, and you are importing a component from a different project?
  - You may need to provide a different `tsconfig` to compodoc. In your storybook builder options you could pass a different tsconfig with `"compodocArgs": ["-e", "json", "-d", ".", "-p", "tsconfig.sb.json"],`.
  
You want to make sure your component is referenced by your `tsconfig` (*basically you want compodoc to consider it part of your project and not from an external project*), so it isn't in `excludes` and is in `include` or `files` or maybe it would also be found if imported by a file in `files`. I don't know if the tsconfig `baseUrl` or `rootDir` affect it also or not. I would have to look into it more to say for sure.
  
If your project has a `src/` and `lib/` folders and the `tsconfig` that Storybook is using has `{ "include": [ "src/**/*.ts" ] }` and your component is at `lib/ex.component.ts` then compodoc is not going to consider that part of your source code, so it won't generate documentation for it.

With more information, I could probably be more specific.
